Sources/GRPC/ConnectionPool/ConnectionPool.swift

```diff
@@ -689,8 +714,14 @@ extension ConnectionPool: ConnectionManagerConnectivityDelegate {
     // Grab the number of reserved streams (before invalidating the index by adding a connection).
     let reservedStreams = self._connections.values[index].reservedStreams
 
+    // If we have less than the minimum number of connections, don't let
+    // the new connection close when idle.
+    let idleBehavior =
+      self.sync.activeConnections < self.minConnections
+      ? ConnectionManager.IdleBehavior.neverGoIdle : .closeWhenIdleTimeout
```

Comment:
Why don't we just get the idle behaviour from connection that closed? We're then always replacing like for like and avoid scanning the state of all connections.

Reply:
Good call.